### PR TITLE
BDOG-816

### PR DESCRIPTION
**Pull request by chotai**

### CODE REVIEW

File: app/uk/gov/hmrc/fileupload/virusscan/ScanningService.scala [hunk 1]
```diff
@@ -28,23 +29,54 @@ object ScanningService {
...
                     (implicit ec: ExecutionContext): Future[ScanResult] = {
     val appendedKey = s3KeyAppender(envelopeId, fileId)
-    getFile(appendedKey, fileRefId.value).flatMap {
+    retries(scan(scanner, getFile(appendedKey, fileRefId.value)), scanTimeoutAttempts, fileId)
+  }
+
+  private def retries(scanResult: => Future[ScanResult],
+                      timeoutAttempts: Int,
+                      fileId: FileId,
+                      acc: Int = 1)
```
Comment (cjwebb) [inline app/uk/gov/hmrc/fileupload/virusscan/ScanningService.scala:60]: Naming is hard, but I'd be tempted to swap the order of these three arguments, and rename two of them, so its maybe something like:

    fileId: FileId,
    maxTimeoutRetries: Int
    currentTimeoutRetry: Int = 1

`acc` is a bit weird IMO. Its normally used in recursion/folds to build a collection of something, but here we're incrementing a number.

File: app/uk/gov/hmrc/fileupload/virusscan/ScanningService.scala [hunk 2]
```diff
@@ -28,23 +29,54 @@ object ScanningService {
...
+                     (implicit ec: ExecutionContext): Future[ScanResult] = {
+    scanResult.flatMap {
+      case result if acc == timeoutAttempts =>
```
Comment (cjwebb) [inline app/uk/gov/hmrc/fileupload/virusscan/ScanningService.scala:63]: As a side-note, I was having slight troubles understanding what happens if I put `numberOfTimeoutAttempts = 0` in the config, just in case we wanted to disable retries.

I *think* it might be fine, as this line in the ScanningService would be 0 == 1 ... and then it would just continue to the default case in the match. I couldn't spot a test for that though - did I miss it, and/or was it something you considered?

Comment (chotai) [reply app/uk/gov/hmrc/fileupload/virusscan/ScanningService.scala:63]: You're correct the last pattern match would kick in and the retry logic would not take effect. It would behave exactly the same way for negative numbers too. 

numberOfTimeoutAttempts in config is optional. When it isn't there, we default to 1 so these really are edge cases, but quick enough to write tests for. 👍

File: app/uk/gov/hmrc/fileupload/virusscan/ScanningService.scala (hunk 1 above)
Comment (JoPintoPaul) [reply app/uk/gov/hmrc/fileupload/virusscan/ScanningService.scala]: Renamed as suggested.

File: app/uk/gov/hmrc/fileupload/virusscan/ScanningService.scala (hunk 2 above)
Comment (JoPintoPaul) [reply app/uk/gov/hmrc/fileupload/virusscan/ScanningService.scala]: Also amended so that's >= rather than ==, in case config is ever set to zero.
